### PR TITLE
Contract in kyc file

### PR DESCRIPTION
**Pull request by lawlawlaw**
1) Contract in KYC file
2) Correctly update unassigned KYC public key set
3) Subscribe to all WHITELIST notifications, even when whitelist is spent to a new address
4) Correct constants.py with the correct WHITELIST initial address

### CODE REVIEW

File: electrum/gui/qt/main_window.py [hunk 1]
```diff
@@ -491,7 +492,11 @@ def init_menubar(self):
         wallet_menu.addAction(_("&Information"), self.show_master_public_keys)
         wallet_menu.addAction(_("&Register"), self.register_wallet_dialog)
         wallet_menu.addAction(_("&Contract"), self.termsandconditions_dialog)
+        wallet_menu.addAction(_("&Onboard Key"), self.dumponboardkey_dialog)
+        wallet_menu.addAction(_("&Rescan Blockchain"), self.rescan_blockchain_dialog)
         wallet_menu.addSeparator()
```
Comment (tomt1664) [inline electrum/gui/qt/main_window.py]: When is this needed?

File: electrum/gui/qt/main_window.py [hunk 2]
```diff
@@ -2711,6 +2711,106 @@ def overwritefile_dialog(self):
...
+            self.wallet.frozen_addresses = set()
+            self.wallet.storage.put('frozen_addresses', [])
+            historyBackup = self.wallet.history
+            self.wallet.clear_history()
+            for it in historyBackup:
+                historyBackup[it] = []
+            self.wallet.history = historyBackup
+            self.wallet.start_threads(self.network)
+            self.network.rescan_blockchain()
+
```
Comment (tomt1664) [inline electrum/gui/qt/main_window.py]: I assume there is no dialog box with rescan blockchain.

File: electrum/network.py [hunk 3]
```diff
@@ -1160,6 +1160,20 @@ def on_notify_header(self, interface, header_dict):
...
+            if os.path.exists(filename):
+                os.remove(filename)
+            offset_path = b.offset_path()
+            if os.path.exists(offset_path):
+                os.remove(offset_path)
+            self.sub_cache = {} 
+            self.blockchains = blockchain.read_blockchains(self.config)
+            self.blockchain_index = 0
+            self.init_headers_file()
+
```
Comment (tomt1664) [inline electrum/network.py]: What does this do - just re-download the headers?

File: electrum/gui/qt/main_window.py (hunk 1 above)
Comment (lawlawlaw) [reply electrum/gui/qt/main_window.py]: Onboard key is required when using multisig with the encrypted whitelisting system. You have to import the onboarding privkey into the other multisig wallets in order to update the address whitelisted status in the wallet. I have removed this for unencrypted whitelist in the next commit.
Rescan blockchain is useful mainly for debugging. It clears the address whitelisting and transaction history and rescans the blockchain for transactions. In the next commit I've added the unassigned_kyc_pubkeys dict to the list of containers that are reset as these are read from the blockchain as well.

File: electrum/gui/qt/main_window.py (hunk 2 above)
Comment (lawlawlaw) [reply electrum/gui/qt/main_window.py]: There is a dialog box, it displays the key, or an error message if needed.

File: electrum/network.py (hunk 3 above)
Comment (lawlawlaw) [reply electrum/network.py]: See reply above.

**Comment by lawlawlaw** [comment]
> This includes Multisig changes from Robertas? Can you close the other PR? Let's get this tested.

Yes, this contains multisig changes by robertas.

**Pull request merged**